src/server/debug/server/database_dump.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+)
+
+func (s *debugServer) collectDatabaseDump(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, 1*time.Minute)
+	defer cancel()
+	if err := s.collectDatabaseStats(ctxWithTimeout, tw, prefix...); err != nil {
+		return err
+	}
+	if err := s.collectDatabaseTables(ctxWithTimeout, tw, prefix...); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *debugServer) collectDatabaseStats(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	queries := map[string]string{
+		"table-sizes": `
+			SELECT nspname AS "schemaname", relname, pg_total_relation_size(C.oid) AS "total_size"
+			FROM pg_class C INNER JOIN pg_namespace N ON (N.oid = C.relnamespace)
+			WHERE nspname NOT IN ('pg_catalog', 'information_schema') AND C.relkind <> 'i' AND nspname !~ '^pg_toast'
+			ORDER BY nspname, relname;
+		`,
+		"activities": `
+			SELECT current_timestamp - query_start as runtime, datname, usename, client_addr, query
+			FROM pg_stat_activity WHERE state != 'idle' ORDER BY runtime DESC;
+		`,
+		"row-counts": `
+			SELECT schemaname, relname, n_live_tup, seq_scan, idx_scan
+			FROM pg_stat_user_tables ORDER BY schemaname, relname;
+		`,
+	}
+	var errs []error
+	for filename, query := range queries {
+		if err := collectDebugFile(tw, filename, "json", func(w io.Writer) error {
+			rows, err := s.database.QueryContext(ctx, query)
+			if err != nil {
+				return errors.EnsureStack(err)
+			}
+			return s.writeRowsToJSON(rows, w)
+		}, prefix...); err != nil {
+			errs = append(errs, errors.Wrapf(err, "failed to query %q", filename))
+		}
+	}
+	if len(errs) > 0 {
+		return fmt.Errorf("%v", errs)
+	}
+	return nil
+}
+
+func (s *debugServer) collectDatabaseTables(ctx context.Context, tw *tar.Writer, prefix ...string) error {
+	tables, err := pachsql.ListTables(ctx, s.database)
+	if err != nil {
+		return errors.EnsureStack(err)
+	}
+	for _, table := range tables {
+		fullPrefix := strings.Join(append(append([]string{}, prefix...), "tables", table.SchemaName), "/")
+		if err := collectDebugFile(tw, table.TableName, "json", func(w io.Writer) error {
+			bufferedWriter := bufio.NewWriter(w)
+			if err := s.collectTable(ctx, bufferedWriter, &table); err != nil {
+				return errors.Wrapf(err, "error collecting table %s.%s", table.SchemaName, table.TableName)
+			}
+			if err := bufferedWriter.Flush(); err != nil {
+				return errors.Wrap(err, "failed to flush buffer")
+			}
+			return nil
+		}, fullPrefix); err != nil {
+			return errors.EnsureStack(err)
+		}
+	}
+	return nil
+}
+
+func (s *debugServer) collectTable(ctx context.Context, w io.Writer, table *pachsql.SchemaTable) error {
+	column, err := pachsql.KeyColumn(ctx, s.database, table)
+	if err != nil {
+		return errors.Wrap(err, "error getting key column")
+	}
+	query := fmt.Sprintf(`SELECT * FROM %s.%s ORDER BY %s ASC`, table.SchemaName, table.TableName, column)
```

Comment:
It's pretty dangerous to interpolate strings that originate from outside of the program directly into SQL queries.  Normally you'd use placeholders, but you can't for table and column names, so you'll have to figure out how to hook in something like https://pkg.go.dev/gopkg.in/jackc/pgx.v1#Connection.QuoteString.

It looks like you can do this:

```
In Go 1.13 and above (*sql.Conn) Raw() can be used to get a *pgx.Conn from
the standard database/sql.DB connection pool. This allows operations that
use pgx specific functionality.

    // Given db is a *sql.DB
    conn, err := db.Conn(context.Background())
    if err != nil {
        // handle error from acquiring connection from DB pool
    }

    err = conn.Raw(func(driverConn interface{}) error {
        conn := driverConn.(*stdlib.Conn).Conn() // conn is a *pgx.Conn
        // Do pgx specific stuff with conn
        conn.CopyFrom(...)
        return nil
    })
    if err != nil {
        // handle error that occurred while using *pgx.Conn
    }
```

I am not sure why our code tries to hide the fact that we use pgx.  It's very annoying in this case.

Reply:
So, I decided to jump to see QuoteString()'s definition and include that since it was pretty simple and call that on the inputs before passing them into the template